jdaviz/configs/cubeviz/plugins/sonify_data/sonify_data.py

```diff
@@ -0,0 +1,118 @@
+
+    sample_rate = IntHandleEmpty(44100).tag(sync=True)
+    buffer_size = IntHandleEmpty(2048).tag(sync=True)
+    assidx = FloatHandleEmpty(2.5).tag(sync=True)
+    ssvidx = FloatHandleEmpty(0.65).tag(sync=True)
+    eln = Bool(False).tag(sync=True)
+    audfrqmin = FloatHandleEmpty(50).tag(sync=True)
+    audfrqmax = FloatHandleEmpty(1500).tag(sync=True)
+    pccut = IntHandleEmpty(20).tag(sync=True)
+    volume = IntHandleEmpty(100).tag(sync=True)
+    stream_active = Bool(True).tag(sync=True)
+    has_strauss = Bool(_has_strauss).tag(sync=True)
+
+    # TODO: can we referesh the list, so sounddevices are up-to-date when dropdown clicked?
+    sound_devices_items = List().tag(sync=True)
+    sound_devices_selected = Unicode('').tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._plugin_description = 'Sonify a data cube'
+        self.docs_description = 'Sonify a data cube using the Strauss package.'
+
+        if self.has_strauss:
+            devices, indexes = self.build_device_lists()
+            self.sound_device_indexes = dict(zip(devices, indexes))
+            self.sound_devices_items = devices
+            self.sound_devices_selected = dict(zip(indexes, devices))[sd.default.device[1]]
+
+        # TODO: Remove hardcoded range and flux viewer
+        self.spec_viewer = self.app.get_viewer('spectrum-viewer')
+        self.flux_viewer = self.app.get_viewer('flux-viewer')
+
+    @property
+    def user_api(self):
+        expose = []
+        return PluginUserApi(self, expose)
+
+    @with_spinner()
+    def vue_sonify_cube(self, *args):
+        # Get index of selected device
+        selected_device_index = self.sound_device_indexes[self.sound_devices_selected]
+        self.flux_viewer.get_sonified_cube(self.sample_rate, self.buffer_size,
+                                           selected_device_index, self.assidx, self.ssvidx,
+                                           self.pccut, self.audfrqmin,
+                                           self.audfrqmax, self.eln)
+
+        # Automatically select spectrum-at-spaxel tool
+        spec_at_spaxel_tool = self.flux_viewer.toolbar.tools['jdaviz:spectrumperspaxel']
+        self.flux_viewer.toolbar.active_tool = spec_at_spaxel_tool
+
+    def vue_start_stop_stream(self, *args):
+        self.stream_active = not self.stream_active
+        self.flux_viewer.stream_active = not self.flux_viewer.stream_active
+
+    @observe('spectral_subset_selected')
+    def update_wavelength_range(self, event):
+        if not hasattr(self, 'spec_viewer'):
+            return
+        display_unit = self.spec_viewer.state.x_display_unit
+        min_wavelength = self.spectral_subset.selected_obj.lower.to_value(u.Unit(display_unit))
+        max_wavelength = self.spectral_subset.selected_obj.upper.to_value(u.Unit(display_unit))
+        self.flux_viewer.update_listener_wls(min_wavelength, max_wavelength, display_unit)
```

Comment:
does this affect an already sonified cube or just the next one?  Why do we have to listen here instead of just pulling this value when clicking "sonify"?

Reply:
It affects the next one, good catch. I moved this code to the `vue_sonify_cube` method.